spec.bs

```diff
@@ -491,19 +509,88 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
-
-    Note: In Chrome's experimentation phase, it will additionally require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> for the request to be eligible for topics.
+          1. Insert |topicItem| to |currentTopicsInnerList|.
+      1. If |currentTopicsInnerList| is not empty:
+          1. Let |topicParameters| be an empty [=Structured Fields Parameters=].
+          1. Set |topicParameters|["<code>v</code>"] to a [=Structured Fields Token=] having words |lastVersion|.
+          1. Associate |topicParameters| with |currentTopicsInnerList|.
+          1. Insert |currentTopicsInnerList| to |topicsStructuredFieldsList|.
+          1. Clear |currentTopicsInnerList|.
+      1. If |numVersionsInEpochs| is 0, then set |numVersionsInEpochs| to 1.
+      1. Let |maxNumberOfEpochs| be 3 (i.e. topics are selected from the last 3 epochs).
+      1. Let |topicMaxLength| be number of base-10 digits in the maximum [=browsing topics types/topic id=] (e.g. for <a href="https://github.com/patcg-individual-drafts/topics/blob/main/taxonomy_v1.md">Chrome's initial taxonomy</a>, |topicMaxLength| is 3, as the [=browsing topics types/topic id=] has maximum 3 digits).
+      1. Let |versionMaxLength| be the length of the current [=browsing topics types/maximum version string length=].
+      1. Let |listItemsSeparatorLength| be 2 (i.e. structured fields use a two characters (", ") to separate list items).
+      1. Let |perVersionedTopicsInnerListOverhead| be 5 (i.e. for "();v=")
+      1. Let |maxPaddingLength| be |maxNumberOfEpochs| * |topicMaxLength| + |maxNumberOfEpochs| - |numVersionsInEpochs| + |numVersionsInEpochs| * |perVersionedTopicsInnerListOverhead| + |numVersionsInEpochs| * |versionMaxLength| + (numVersionsInEpochs - 1) * |listItemsSeparatorLength|.
+      1. Let |paddingLength| be |maxPaddingLength|.
+      1. If |topicsStructuredFieldsList| is not empty:
+          1. Let |serializedTopicsList| be the result of executing the <a href="https://httpwg.org/specs/rfc8941.html#text-serialize">serializing structured fields</a> algorithm on |topicsStructuredFieldsList|.
+          1. Decrement |paddingLength| by |serializedTopicsList|'s length.
+      1. Else:
+          1. Increment |paddingLength| by |listItemsSeparatorLength| (i.e. to account for the separator characters that would be added when |topics| are not empty).
+      1. If |paddingLength| < 0, then set |paddingLength| to 0.
+
+      Note: the padding should generally be >= 0. It may be negative in certain circumstances: when historically stored topic versions are greater (and use more digits) than the current [=browsing topics types/maximum version string length=]; or when there is a race between getting topics and getting the number of distinct topic versions. Clamp to 0 to prevent breakage in these rare circumstances.
```

Comment:
```suggestion
      Note: the padding should generally be &geq; 0. It may be negative in certain circumstances: when historically stored topic versions are greater (and use more digits) than the current [=browsing topics types/maximum version string length=]; or when there is a race between getting topics and getting the number of distinct topic versions. Clamp to 0 to prevent breakage in these rare circumstances.
```